src/main/java/org/opensearch/sdk/BaseExtensionRestHandler.java

```diff
@@ -21,6 +21,7 @@
 import org.opensearch.extensions.rest.ExtensionRestRequest;
```

Comment:
Do we need this import/file anymore?

Reply:
Nope, it was in a stray javadoc so didn't get auto-cleaned by spotless like everything else.

Reply:
What about `ExtensionRestRequest`file? Since we are using `RestRequest` should we get delete it as a part of this PR?

Reply:
No, we still need that as part of transport, we're just removing the "dual use" nature of it.

Reply:
We still need some parts of it, particularly the parsed `params()` that were generated by `RestController` using the registered rest action named params.

Reply:
Shouldn't we document the usage of both the requests then? When to use `ExtensionRestRequest` and `RestRequest` in the migration doc?

Reply:
For migration, using "RestRequest" isn't a change at all, it will minimize the diff. And the ExtensionRestRequest is documented [here](https://github.com/opensearch-project/opensearch-sdk-java/blob/main/PLUGIN_MIGRATION.md#replace-bytesrestresponse-with-extensionrestresponse).  (The lines above it are being removed as part of this PR.)

Reply:
Anything documented about `ExtensionRestRequest` or plugins won't need it at all for the migration and it will be required only by SDK moving forward?

Reply:
> Anything documented about `ExtensionRestRequest` or plugins won't need it at all for the migration and it will be required only by SDK moving forward?

That is correct.  The sequence:
 - user submits a HttpRequest
 - Rest Controller parses this into a RestRequest including named params (tricky!)
 - we use ExtensionRestRequest to extract the parts we need out of RestRequest and send to SDK
 - SDK reassembles a new RestRequest which is mostly like the old one and is further used by extensions without any change to existing code
 - The only tricky thing needed is that users can't just send back a BytesRestResponse like usual; that is wrapped in the ExtensionRestResponse in order to track the "consumed parameters" of the request.

Reply:
> 1. In the future can we eliminate the ExtensionRestRequest completely and use RestRequest since we will be having security for the headers too? This will help us to ease the pain of creating another RestRequest in SDK

We need some method of passing the method, uri, content type headers, content bytes, and params over transport.  You can call it whatever you want, and maybe even change to protobuf, but those bits need to get from point A to point B somehow.

> 2. Again, in the future can we just send back BytesRestResponse and parse it in SDK to eliminate `ExtensionRestResponse`?

You need some way of communicating "consumed params" along with the bytes or, if you want to just ignore the whole consumed params thing, you could.